server/supabase/migrations/20240607225621_add_query_row.sql

```diff
@@ -0,0 +1,54 @@
+create type status as enum ('QUEUED', 'STARTING', 'COMPILING', 'WAITING_TO_START', 'IN_PROGRESS', 'COMPLETE', 'KILLED', 'NOT_FOUND', 'CRASHED', 'UNKNOWN');
+
+create type query_type as enum ('IPA', 'DEMO_LOGGER');
+
+create table
+queries (
+uuid uuid default gen_random_uuid() primary key,
+display_id varchar(255) unique not null,
+type query_type not null,
+status status not null,
+form_data jsonb not null default '{}'::jsonb,
+created_at timestamp default current_timestamp not null,
+started_at timestamp,
+ended_at timestamp
+);
+
+create index idx_display_id on queries (display_id);
+
+alter table queries enable row level security;
+
+create policy "Queries are visible to authenticated users"
+on queries for select
+to authenticated
+using ( true );
+
+create policy "Queries are only created by authenticated users"
+on queries for insert
+to authenticated
+with check ( true );
+
+create policy "Queries are only updated by authenticated users"
+on queries for update
+to authenticated
+using ( true )
+with check ( true );
+
+create or replace function generate_unique_display_id(p_display_id varchar) returns varchar as $$
```

Comment:
My primary motivation here was to avoid long urls (e.g., draft.test/query/d01a7f44-4697-4bec-9658-33300118e532) with the uuid of the query, as well as adding something to mentally separate queries.

If we are at all concerned about collision, I strongly prefer not moving to the application layer, as that adds an additional round trip to the database. 

To avoid the concern where this gets worse over time (e.g., risk of conflicts increases as more queries are created), I like the idea of adding the timestamp (though keeping it short, so maybe just YYYYMMDDHHMM.) if the url is draft.test/query/adjetive-noun-YYMMDDHHMM, that's only a bit longer than today (and actually a bit more meaningful), and should be very hard to cause any collision.

Reply:
ok, so updated in client logic. I still prefer to keep this here as a failsafe, as it will just append some extra digits to the end if there happen to be massive amounts of queries coming in within the same minute.